vcdriver/helpers.py

```diff
@@ -112,19 +112,25 @@ def timeout_loop(
 
     :raise: TimeoutError: If the timeout is reached
     """
+    error = None
     if not quiet:
         print('Waiting for [{}] ... '.format(description), end='')
         sys.stdout.flush()
     countdown = timeout
     start = time.time()
     while countdown >= 0:
         callback_start = time.time()
-        if callback(*callback_args, **callback_kwargs):
-            break
+        try:
+            if callback(*callback_args, **callback_kwargs):
+                break
+        except Exception as e:
+            error = e
```

Comment:
Just an observation: Only last error will be shown if more than one exception occurs during timeout period.
If error is the same, most likely, this is not an issue and will also prevent description being flooded with the same  message.

Reply:
The problem with it is that you might have thousands of attempts within a single timeout depending on the poll interval and the actual timeout, so I thought it was sensible to pickup the last one, which will be the one you're after in 99% of the cases :+1: